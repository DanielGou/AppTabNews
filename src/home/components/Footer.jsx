import { ActivityIndicator, View, StyleSheet } from 'react-native';

export function Footer(isLoading) {
    return () => {
        if (isLoading)
            return null;
        return (
            <View style={styles.footerLoading}>
                <ActivityIndicator size="large" />
            </View>
        );
    };
}

export const styles = StyleSheet.create({
    footerLoading:{
        margin: 20
    }
});
