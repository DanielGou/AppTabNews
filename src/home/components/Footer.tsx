import React from 'react';
import { ActivityIndicator, View, StyleSheet } from 'react-native';

export function Footer({ isLoading }: { isLoading: any; }) {
    if (!isLoading){
        return (
            <View style={styles.footerLoading}>
                <ActivityIndicator size="large" />
            </View>
        );
    }
}

export const styles = StyleSheet.create({
    footerLoading:{
        margin: 20
    }
});
