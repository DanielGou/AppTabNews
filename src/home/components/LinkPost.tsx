import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function LinkPost({ post }: { post: any; }): JSX.Element{
    const { title, username, children_deep_count, publishedAt } = post.item
 
    return(
        <View style={styles.wrapperLinkPost}>
            <Text style={styles.title}>{post.index + 1}.</Text>
            <View style={styles.ml5}>
                <Text style={styles.title}>{title}</Text>
                <View style={styles.wrapperInfoPost}>
                    <Text>{children_deep_count} comentários  ·</Text>
                    <Text style={styles.ml5}>{username}  ·</Text>
                    <Text style={styles.ml5}>{publishedAt}</Text>
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    wrapperLinkPost:{
        display: 'flex',
        flexDirection: 'row',
        margin: 15,
    },
    title:{
        fontSize: 17,
        fontWeight: 'bold'
    },
    ml5:{
        marginLeft: 5
    },
    wrapperInfoPost:{
        display: 'flex',
        flexDirection: 'row',
    }
})