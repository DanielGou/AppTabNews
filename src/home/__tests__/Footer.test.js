import React from "react";
import { create } from 'react-test-renderer';

import { Footer } from "../components/Footer";


const tree = create(Footer(false))

test('snapshot Home', ()=>{
    expect(tree).toMatchSnapshot()
})
