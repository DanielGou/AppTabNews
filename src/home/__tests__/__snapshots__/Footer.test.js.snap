// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Home 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
